src/workerd/api/actor-alarms-delete-test.js

```diff
@@ -0,0 +1,74 @@
+import * as assert from 'node:assert'
+
+async function timeout(ms) {
+  await scheduler.wait(ms);
+  throw new Error(`timed out`);
+}
+
+export class DurableObjectExample {
+  constructor(state) {
+    this.state = state;
+  }
+
+  async waitForAlarm(scheduledTime, timeoutOverrideMs) {
+    console.log(`waiting for ${scheduledTime.valueOf()}`);
+    let self = this;
+    let prom = new Promise((resolve) => {
+      self.resolve = resolve;
+    });
+
+    let timeMs = scheduledTime.valueOf();
+    let timeoutMs = (timeMs - Date.now().valueOf()) + (timeoutOverrideMs ?? 4000);
+    try {
+      await Promise.race([prom, timeout(timeoutMs)]);
+      if (Date.now() < scheduledTime.valueOf()) {
+        throw new Error(`Date.now() is before scheduledTime! ${Date.now()} vs ${scheduledTime.valueOf()}`);
+      }
+    } catch (e) {
+      throw new Error(`error waiting for alarm at ${scheduledTime.valueOf()}: ${e}`);
+    }
+
+
+    let alarm = await this.state.storage.getAlarm();
+    if (alarm != null) {
+      throw new Error(`alarm time not cleared when handler ends. ${alarm}`);
+    }
+  }
+
+  async alarm() {
+    console.log("alarm()");
+    let time = await this.state.storage.getAlarm();
+    if (time) {
+      throw new Error(`time not null inside alarm handler ${time}`);
+    }
+    // Used to crash here
+    await this.state.storage.deleteAlarm();
+    console.log("success!");
+    this.resolve();
+  }
+
+  async fetch(request) {
+    await this.state.storage.setAlarm(500);
```

Comment:
nit: This is verifying that set and delete works fine, right? Mind dropping a comment explaining?